### PR TITLE
SslStream - update documentation

### PR DESCRIPTION
**Pull request by aik-jahoda**
## Summary

Collect all changes from #3916 related to SslStream

### CODE REVIEW

**Comment by aik-jahoda** [comment]
Doc preview: https://review.docs.microsoft.com/en-us/dotnet/api/system.net.security.sslstream?branch=pr-en-us-4372&view=netcore-3.1

File: xml/System.Net.Security/SslStream.xml [hunk 1]
```diff
@@ -3441,7 +3490,7 @@ Authentication has not occurred.</exception>
         <ReturnType>System.Net.Security.SslApplicationProtocol</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>The TLS Application protocol which is negotiated after this <see cref="T:System.Net.Security.SslStream" /> authenticate.</summary>
```
Comment (ManickaP) [inline xml/System.Net.Security/SslStream.xml:3493]:
```suggestion
        <summary>The negotiated application protocol in TLS handshake.</summary>
```

File: xml/System.Net.Security/SslStream.xml [hunk 2]
```diff
@@ -3441,7 +3490,7 @@ Authentication has not occurred.</exception>
...
         <value>To be added.</value>
         <remarks>To be added.</remarks>
```
Comment (ManickaP) [inline xml/System.Net.Security/SslStream.xml:3495]:
```suggestion
        <remarks>Available once authentication via one of of the AuthenticateAs... methods has been successfully processed.</remarks>
```
Probably worth adding links to `AuthenticateAsClient/Server`.

Also it comes from [ALPN](https://tools.ietf.org/html/rfc7301) which might be worth mentioning, or not.

File: xml/System.Net.Security/SslStream.xml [hunk 3]
```diff
@@ -2828,10 +2870,17 @@ Authentication has not occurred.</exception>
...
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        <summary>Asynchronously writes any buffered data to the underlying device.</summary>
+        <returns>A task that represents the asynchronous flush operation.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+ This method invokes <xref:System.IO.Stream.Flush%2A> on the underlying stream.
```
[the remark this answers is not in the document]
Comment (aik-jahoda) [reply xml/System.Net.Security/SslStream.xml]: Agree, we probably need an issue for enhance examples as it requires VB + C++ examples update too.

**Pull request merged**